internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go

```diff
@@ -192,11 +194,23 @@ func newJWTAuthenticator(spec *auth1alpha1.JWTAuthenticatorSpec) (*jwtAuthentica
 		return nil, fmt.Errorf("issuer %q does not have jwks_uri set", spec.Issuer)
 	}
 	oidcAuthenticator, err := oidc.New(oidc.Options{
-		IssuerURL:            spec.Issuer,
+		JWTAuthenticator: apiserver.JWTAuthenticator{
+			Issuer: apiserver.Issuer{
+				URL:       spec.Issuer,
+				Audiences: []string{spec.Audience},
+			},
+			ClaimMappings: apiserver.ClaimMappings{
+				Username: apiserver.PrefixedClaimOrExpression{
+					Claim:  usernameClaim,
+					Prefix: ptr.To(""),
+				},
+				Groups: apiserver.PrefixedClaimOrExpression{
+					Claim:  groupsClaim,
+					Prefix: ptr.To(""),
+				},
+			},
+		},
```

Comment:
The Kube libs refactored `oidc.Options` dramatically in v0.29. This change deserves very careful review to ensure that our behavior will not change compared to before. We should not merge until we have reviewed this closely.

Reply:
@benjaminapetersen and I reviewed the old implementation of `oidc.New(oidc.Options(...))` and we found that the behavior should be identical. Both the old code and the new code validates that the `aud` claim of the ID token includes the client ID string (called `spec.Audience` in our code above).